shell/pages/c/_cluster/fleet/index.vue

```diff
@@ -284,223 +425,510 @@ export default {
+                :selected-states="stateFilter[workspace.id] || {}"
+                @click:state="selectStates(workspace.id, $event)"
+              />
+              <ResourcePanel
+                v-if="workspace.clusters?.length"
+                :data-testid="'resource-panel-clusters'"
+                :states="clusterStates[workspace.id]"
+                :workspace="workspace.id"
+                :type="FLEET.CLUSTER"
+                :selectable="false"
               />
-            </template>
-            <template #cell:bundlesReady="{row}">
-              <span v-if="ws.type === 'namespace'"> - </span>
-              <CompoundStatusBadge
-                v-else
-                data-testid="bundles-ready"
-                :tooltip-text="getTooltipInfo('bundles', row, gitReposCounts)"
-                :badge-class="getStatusInfo('bundles', row, gitReposCounts).badgeClass"
-                :icon="getStatusInfo('bundles', row, gitReposCounts).icon"
-                :value="getBadgeValue('bundles', row, gitReposCounts)"
+              <ResourcePanel
+                v-if="workspace.clusterGroups?.length"
+                :data-testid="'resource-panel-cluster-groups'"
+                :states="clusterGroupsStates[workspace.id]"
+                :workspace="workspace.id"
+                :type="FLEET.CLUSTER_GROUP"
+                :show-chart="false"
+                :selectable="false"
               />
-            </template>
-            <template #cell:resourcesReady="{row}">
-              <CompoundStatusBadge
-                data-testid="resources-ready"
-                :tooltip-text="getTooltipInfo('resources', row, gitReposCounts)"
-                :badge-class="getStatusInfo('resources', row, gitReposCounts).badgeClass"
-                :icon="getStatusInfo('resources', row, gitReposCounts).icon"
-                :value="getBadgeValue('resources', row, gitReposCounts)"
+            </div>
+          </div>
+          <div class="card-panel-main-actions">
+            <div
+              v-if="workspace.repos?.length"
+              class="expand-button"
+              :data-testid="'expand-button'"
+            >
+              <i
+                role="button"
+                tabindex="0"
+                :aria-label="t(`${ isWorkspaceCollapsed[workspace.id] ? 'expand': 'collapse' }-${ workspace.id }`)"
+                :class="{
+                  ['icon icon-lg icon-chevron-right']: isWorkspaceCollapsed[workspace.id],
+                  ['icon icon-lg icon-chevron-down']: !isWorkspaceCollapsed[workspace.id],
+                }"
+                @click="toggleCard(workspace.id)"
+                @keydown.space.enter.stop.prevent="toggleCard(workspace.id)"
               />
-            </template>
-
-            <template #cell:target="{row}">
-              {{ row.targetInfo.modeDisplay }}
-            </template>
-          </ResourceTable>
-        </template>
-      </CollapsibleCard>
+            </div>
```

Comment:
Is there anything that would prevent us from using the button component? We can get keyboard behaviors and accessibility options out of the box without the needing to manually add all of this to the icon:

```js
<RcButton 
  :aria-label="t(`${ isWorkspaceCollapsed[workspace.id] ? 'expand': 'collapse' }-${ workspace.id }`)"
  @click="toggleCard(workspace.id)"
>
  <i
    :class="{
      ['icon icon-lg icon-chevron-right']: isWorkspaceCollapsed[workspace.id],
      ['icon icon-lg icon-chevron-down']: !isWorkspaceCollapsed[workspace.id],
    }"
  />
</RcButton>
```